File: fc_test.py
import ROOT as rt
import numpy as np
import time


def main():
    fc = rt.TFeldmanCousins()
    fc.SetCL(.9)
    fc.SetMuMax(50)
    fc.SetMuStep(0.0005)

    n_obs = np.random.random(1000) * 5
    n_bg = np.random.random(1000) * 10

    t0 = time.time()
    for i in range(1000):
        fc.CalculateUpperLimit(n_obs[i], n_bg[i])

    t1 = time.time()

    print(f"TIME {t1 - t0} s")
    return


main()

File: fc_test.cpp
std::vector<float> aRandomVector(double maxValue, int n)
{
    std::vector<float> result;
    std::srand(std::time(nullptr)); // use current time as seed for random generator
    int random_value = std::rand();
    for (int i = 0; i < n; i++)
    {
        result.emplace_back(std::rand() / (RAND_MAX + 1u) * maxValue);
    }
    return result;
}

void fc()
{
	auto observed = aRandomVector(5.0, 1000);
	auto background = aRandomVector(10.0, 1000);

	auto fc = new TFeldmanCousins();
	fc->SetCL(.9);
    fc->SetMuMax(50);
    fc->SetMuStep(0.0005);

    std::time_t result1 = std::time(nullptr);

	for (int i = 0; i < 1000; i++)
	{
        fc->CalculateUpperLimit(observed[i], background[i]);
	}

	std::time_t result2 = std::time(nullptr);

	std::cout << result2 - result1 << std::endl;
}
